### PR TITLE
add resuse_client option

### PR DESCRIPTION
**Pull request by logan-markewich**
# Description

Frequent LLM/Embddings clients occur during async operation. Hypothesis is that this is due to sharing a single OpenAI client across all API calls.

Fix is to add an option that lets a new client be created for each API call

## Type of Change

- [x] Bug fix (non-breaking change which fixes an issue)

### CODE REVIEW

File: llama_index/embeddings/openai.py [hunk 1]
```diff
@@ -280,25 +288,36 @@ def __init__(
...
+    def _get_client(self) -> OpenAI:
+        if not self.reuse_client:
+            return OpenAI(**self._get_credential_kwargs())
+
+        if self._client is None:
+            self._client = OpenAI(**self._get_credential_kwargs())
+        return self._client
+
+    def _get_aclient(self) -> AsyncOpenAI:
+        if not self.reuse_client:
```
Comment (nerdai) [inline llama_index/embeddings/openai.py:315]: Should this be "if self.reuse_client:" (then renew the client?)

File: llama_index/llms/azure_openai.py [hunk 2]
```diff
@@ -63,6 +63,7 @@ def __init__(
         additional_kwargs: Optional[Dict[str, Any]] = None,
         max_retries: int = 3,
         timeout: float = 60.0,
+        reuse_client: bool = False,
```
Comment (nerdai) [inline llama_index/llms/azure_openai.py:66]: I think default is True elsewhere. Should this also be true?

File: llama_index/embeddings/openai.py (hunk 1 above)
Comment (nerdai) [reply llama_index/embeddings/openai.py:315]: Nvm lol. I think it's right. I got my negatives in wrong order 😂

**Review by nerdai — APPROVED** [review]
LGTM! just a minor question re: a potential inconsistency in default value of reuse_client

File: llama_index/llms/azure_openai.py (hunk 2 above)
Comment (logan-markewich) [reply llama_index/llms/azure_openai.py:66]: Good catch -- copilot autocomplete strikes again lol

**Pull request merged**